src/components/Shared/MessagesList.tsx

```diff
@@ -0,0 +1,123 @@
+import getAvatar from '@lib/getAvatar';
+import { Message } from '@xmtp/xmtp-js';
+import React, { ReactNode } from 'react';
+import { useAppStore } from 'src/store/app';
+
+export type MessageListProps = {
+  messages: Message[];
+};
+
+type MessageTileProps = {
+  message: Message;
+};
+
+interface Props {
+  children: ReactNode;
+}
+
+const formatTime = (d: Date | undefined): string =>
+  d
+    ? `${d.toLocaleTimeString(undefined, {
+        hour12: true,
+        hour: 'numeric',
+        minute: '2-digit'
+      })} - ${d.toLocaleDateString()}`
+    : '';
+
+const isOnSameDay = (d1?: Date, d2?: Date): boolean => {
+  return d1?.toDateString() === d2?.toDateString();
+};
+
+const formatDate = (d?: Date) =>
+  d?.toLocaleString('en-US', { year: 'numeric', month: 'long', day: 'numeric' });
+
+const MessageTile = ({ message }: MessageTileProps): JSX.Element => {
```

Comment:
**suggestion:** you can use something like this

```
const MessageTile: FC<MessageTileProps> = ({ message }) => {
```

https://github.com/lensterxyz/lenster/blob/main/src/components/Home/Hero.tsx#L4